src/request.rs

```diff
@@ -610,6 +637,38 @@ mod tests {
         assert!(tracker.borrow().dropped);
     }
 
+    #[actix_rt::test]
+    async fn test_path_routes() {
+        let mut srv = init_service(
+            App::new().service(
+                web::scope("/user/{id}")
+                    .service(web::resource("/profile").route(web::get().to(
+                        move |req: HttpRequest| {
+                            assert_eq!(
+                                req.0.rmap.match_name("/user/22/profile"),
+                                Some(ResourceDef::new("/profile").name().to_owned())
+                            );
```

Comment:
Again here, we need to set .name("x") on the resource and assert that req.match_name().unwrap() equals "x", right?